docstrings: Change single dash to double dash after input variables

gmou3 opened the pull request:
This is achieved by using variants of the following regex rule:
```
Find: ([ ]*- [^ *()\[\]^]* )-( )
Replace: $1--$2
```

**For the reviewer**: to view the _diff_ succinctly run
`git diff 10.4.beta7 -U0 | grep '^[+-]' | grep -Ev '^(--- a/|\+\+\+ b/)'`
or see the attached file: [diff.txt](https://github.com/sagemath/sage/files/15447061/diff.txt) (computed from 307cbf9).

Similar command with colors:
`git diff --color 10.4.beta7 -U0 | grep -Ev 'diff|index|@@|---|\+\+\+'`

DavidAyotte committed:
initial working implementation

DavidAyotte committed:
implement a rank two specific class

DavidAyotte committed:
delete some unused import

DavidAyotte committed:
add some parameter checks for DrinfeldModularForms

DavidAyotte committed:
minor change in example

DavidAyotte committed:
add some doctests

DavidAyotte committed:
add some placeholder methods

DavidAyotte committed:
add some doctests

DavidAyotte committed:
fix a bug an add some doctests

DavidAyotte committed:
add some doctests to coefficient_form

DavidAyotte committed:
refactor coefficient_forms method

DavidAyotte committed:
add placeholder method

DavidAyotte committed:
some details

DavidAyotte committed:
add an introduction to Drinfeld modular forms

DavidAyotte committed:
apply E228

DavidAyotte committed:
fix EXAMPLES block

DavidAyotte committed:
fix repeated word

DavidAyotte committed:
refactor the default names creation

DavidAyotte committed:
remove rank 2 specific classes

DavidAyotte committed:
fix failing doctests

DavidAyotte committed:
small fixes

DavidAyotte committed:
add drinfeld modforms to modular forms toc index

fchapoton committed:
Update ring.py

fixing the linter warning

DavidAyotte committed:
add some backticks

DavidAyotte committed:
Merge branch 'drinfeld_modform' of https://github.com/DavidAyotte/sage into drinfeld_modform

DavidAyotte committed:
add details to DrinfeldModularFormsElement documentation

DavidAyotte committed:
rename is_drinfeld_modular_forms into is_homogeneous and type_m into type

DavidAyotte committed:
use super

DavidAyotte committed:
change some phrasing

DavidAyotte committed:
add a note to the rank method

DavidAyotte committed:
move top level documentation inside the main class

DavidAyotte committed:
some documentation rephrasing

DavidAyotte committed:
add documentation to DrinfeldModularForms main class

DavidAyotte committed:
fix some failing doctests

DavidAyotte committed:
remove top level examples in ring.py

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

Commit:
build/pkgs/patch: Update to 2.7.6, add upstream_url

Commit:
build/pkgs/patch: Use updated build-aux scripts

Commit:
build/pkgs/m4ri: Update to 20200125

Commit:
build/pkgs/m4rie: Update to 20200125

Commit:
build/pkgs/{cliquer,gengetopt,givaro,libgd,lrcalc}: Use newer version of build-aux

Commit:
build/pkgs/m4ri/patches/mzd_fix.patch: Remove, upstreamed

Commit:
build/pkgs/{libhomfly,ppl,rw,symmetrica,zeromq}: Use newer version of build-aux

Commit:
build/pkgs/{m4ri,m4rie}: Use newer version of build-aux

Commit:
build/pkgs/{ntl,brial}: Use newer version of build-aux

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: minor doc change and add doctest for coverage

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: some doc changes

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: address review comments

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: add a reference and a SageMath implementation rubric

DavidAyotte committed:
modular/drinfeld_modform/ring.py: some documenation update, change 'the ring' to 'this ring'

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: small docs changes

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add backticks

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: run TestSuite when q is not prime

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add a missing doctest

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: use Antoine's better code

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: enhance names parsing

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add some doctests

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: fix failing doctests

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: adjust the docstring of DrinfeldModularForms

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
sage/modular/drinfeld_modform/ring.py: add missing doctests and refactor coefficient form(s) methods

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/doc/en/reference/references/index.rst: add Gekeler's reference

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: reference Gekeler's paper

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: fix typo

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: TypeError -> NotImplementedError for invalid operator comparison

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: fix typos

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add documention for coverage

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: allow to pass a prefix for names of the generators

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add documentation on names creation

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: enhance error message

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add a reference block in the DrinfeldModularForms class docstring

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: fix sphinx doc references

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: cross reference element class in parent class docstring

DavidAyotte committed:
Merge branch 'drinfeld_modform' of https://github.com/DavidAyotte/sage into drinfeld_modform

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add rank condition when parsing names

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: fix input checks in coefficient forms methods

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/sage/modular/all.py: fix relative import

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: add a note on the shift of indexing

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: doc line lengths fix

DavidAyotte committed:
fix the type confusion

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: TypeError -> NotImplementedError

tscrim committed:
Implement all G(r,p,m) complex reflection groups.

Commit:
.github/workflows/macos.yml: Update for changed macOS runners

Commit:
.github/workflows/ci-macos.yml (local-macos-nohomebrew): Remove macos-11, add macos-13

ComboProblem committed:
added tests for copy method and verified correct functionality

mkoeppe committed:
LPDictionary.__copy__: Add explanation, blank line before tests

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/doc/en/reference/references/index.rst: fix documentation build fail

DavidAyotte committed:
fix docbuild error (hopefully)

DavidAyotte committed:
change email

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: use the commutative category of graded algebras

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: fix type documentation

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: various documentation fixes

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: some doc fixes

DavidAyotte committed:
src/sage/modular/drinfeld_modform/tutorial.py: small fixes

tscrim committed:
Adding some more documentation and changing the name.

mkoeppe committed:
build/pkgs/ecl: Update to 24.5.10

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

Commit:
Merge branch 'update_hashes' into macos-arm64-upgrades

Commit:
sage -package fix-checksum :all:

mkoeppe committed:
for a in "CLASS='FSMState|FSMTransition|FiniteStateMachine|Automaton|Transducer|FSMProcessIterator';SUFFIX="; do eval "$a"; git --no-pager grep -l -E 'is_('$CLASS')' | xargs sed -E -i.bak 's/import is_('$CLASS')/import \1'$SUFFIX'/;/^ +/s/is_('$CLASS')[(]([^()]*([(][^()]*[)])?[^()]*)[)]/isinstance(\2, \1'$SUFFIX')/g'; done

mkoeppe committed:
reverts

mkoeppe committed:
src/sage/combinat/finite_state_machine.py: Deprecate is_... functions

mkoeppe committed:
fixup isinstance

mkoeppe committed:
reverts

mkoeppe committed:
deprecations

mkoeppe committed:
src/sage/numerical/interactive_simplex_method.py: Rephrase comment

gmou3 committed:
Merge remote-tracking branch 'origin/flats_matroid' into setsystem_input

gmou3 committed:
`SetSystem`: Minor change to accomodate set input

This change makes the code cleaner in multiple places.

mkoeppe committed:
src/sage/ext/fast_callable.pyx: Docstring cosmetics

mkoeppe committed:
sage.rings: Modularization fixes for singular

mkoeppe committed:
sage.rings: Modularization fixes (imports)

mkoeppe committed:
build/pkgs/{m4ri,m4rie}/checksums.ini: Add sha256

mkoeppe committed:
sage.graphs: Doctest cosmetics

mkoeppe committed:
sage.geometry: Modularization fixes (imports), # needs

mkoeppe committed:
src/conftest.py: Update for pytest 8.2.1

mkoeppe committed:
src/tox.ini (pytest): Use consider_namespace_packages = True

tscrim committed:
Renaming class and changing entry point. Fixing typo.

mkoeppe committed:
src/sage/geometry/hyperplane_arrangement/library.py: Fix import

mkoeppe committed:
.ci/write-dockerfile.sh: Quoting fix

mkoeppe committed:
Apply suggestions from code review

Co-authored-by: gmou3 <32706872+gmou3@users.noreply.github.com>

mkoeppe committed:
Apply suggestions from code review

Co-authored-by: gmou3 <32706872+gmou3@users.noreply.github.com>

mkoeppe committed:
Apply suggestions from code review

Co-authored-by: gmou3 <32706872+gmou3@users.noreply.github.com>

mkoeppe committed:
Apply suggestions from code review

Co-authored-by: gmou3 <32706872+gmou3@users.noreply.github.com>

gmou3 committed:
Few more docstring edits

gmou3 committed:
Change single dash to double dash after input vars

Using the following regex:
Find: (
[ ]*- [^ *()\[\]^]* )-( )
Replace: $1--$2

gmou3 committed:
Double dashes after one-line multiple input vars

gmou3 committed:
Removing few unintended dashes

mkoeppe committed:
git --no-pager grep -l 'deprecation(' src/sage/modular | xargs sed -i.bak '/ is deprecated; use /s/the function /The function /;/ is deprecated; use /s/ instead/ instead./'

mkoeppe committed:
git --no-pager grep -l 'deprecation(' src/sage/combinat/finite_state_machine.py | xargs sed -i.bak '/ is deprecated; use /s/the function /The function /;/ is deprecated; use /s/ instead/ instead./'

gmou3 committed:
Reseparate random and long time

mkoeppe committed:
src/sage/ext/fast_callable.pyx (get_builtin_functions): Update docstring

mkoeppe committed:
src/sage/rings/polynomial/polynomial_singular_interface.py (_do_singular_init_): Remove unnecessary handling of 'singular is None'

mkoeppe committed:
src/sage/rings/polynomial/polynomial_singular_interface.py (PolynomialRing_singular_repr._singular_init_): Handle 'singular is None' here

mkoeppe committed:
src/sage/rings/polynomial/polynomial_singular_interface.py: Remove global variable 'singular'

mkoeppe committed:
PolynomialRing_singular_repr._singular_init_: Reorder for clarity

kwankyu commented on src/sage/combinat/words/word_infinite_datatypes.py:
```diff
@@ -26,7 +26,7 @@ def __init__(self, parent, callable, length=None):
+        - ``parent`` -- a parent
         -  ``callable`` - a callable defined on ``range(stop=length)``
         -  ``length`` - (default: ``None``) nonnegative integer or ``None``
 
```
lines 30,31?

kwankyu commented on src/sage/combinat/words/word_infinite_datatypes.py:
```diff
@@ -589,7 +589,7 @@ def __init__(self, parent, iter, length=None):
         INPUT:
 
-        - ``parent`` - a parent
+        - ``parent`` -- a parent
```
The following lines too?

kwankyu commented on src/sage/combinat/words/word_infinite_datatypes.py:
```diff
@@ -922,7 +922,7 @@ def __init__(self, parent, iter, length=None):
         INPUT:
 
-        - ``parent`` - a parent
+        - ``parent`` -- a parent
```
The same here?

kwankyu commented on src/sage/crypto/lwe.py:
```diff
@@ -253,17 +253,17 @@ def __init__(self, n, q, D, secret_dist='uniform', m=None):
+          - "uniform" -- secret follows the uniform distribution in `\Zmod{q}`
+          - "noise" -- secret follows the noise distribution
           - ``(lb,ub)`` - the secret is chosen uniformly from ``[lb,...,ub]`` including both endpoints
 
```
line 264 too?

kwankyu commented on src/sage/matrix/matrix_double_dense.pyx:
```diff
@@ -2220,11 +2220,11 @@ cdef class Matrix_double_dense(Matrix_numpy_dense):
 
-        - ``tol`` - default: ``1e-12`` - the largest value of the
+        - ``tol`` -- default: ``1e-12`` - the largest value of the
           absolute value of the difference between two matrix entries
```
The changes in this file need more touch. For  this instance, 
```
- ``tol`` -- (default: ``1e-12``) the largest value of the ...
```

kwankyu commented on src/sage/matrix/special.py:
```diff
@@ -685,19 +685,19 @@ def diagonal_matrix(arg0=None, arg1=None, arg2=None, sparse=True):
 
-    - ``sparse`` - default: ``True`` - whether or not
+    - ``sparse`` -- default: ``True`` - whether or not
       the result has a sparse implementation.
```
```
- ``sparse`` -- (default: ``True``) whether or not ...
```
more similar changes in this file.

kwankyu commented on src/sage/misc/html.py:
```diff
@@ -269,19 +269,19 @@ def eval(self, x, globals=None, locals=None, mode='display', combine_all=False):
         -  ``locals`` - extra local variables used when
            evaluating Sage code in ``x``.
 
-        - ``mode`` - string (optional, default ``'display'``):
```
Above lines 274, 277?

kwankyu commented on src/sage/misc/sage_unittest.py:
```diff
@@ -175,7 +175,7 @@ def run(self, category=None, skip=[], catch=True, raise_on_failure=False,
-         - ``raise_on_failure`` - a boolean (default: False)
+         - ``raise_on_failure`` -- a boolean (default: False)
          - ``catch``            - a boolean (default: True)
 
```
Other lines?

gmou3 committed:
More double dashes

Search for similar patterns with double spaces, e.g.:
"    -  something - "

kwankyu commented on src/sage/modular/modsym/p1list.pyx:
```diff
@@ -49,7 +49,7 @@ cdef int c_p1_normalize_int(int N, int u, int v,
 
-    - ``ss`` - an integer such that `(ss*uu, ss*vv)` is congruent to
+    - ``ss`` -- an integer such that `(ss*uu, ss*vv)` is congruent to
        `(u,v)` (mod `N`);
```
line 50?

many more below.

kwankyu commented on src/sage/modular/pollack_stevens/dist.pyx:
```diff
@@ -65,7 +65,7 @@ def get_dist_classes(p, prec_cap, base, symk, implementation):
 
-    - ``implementation`` - string - If not None, override the
+    - ``implementation`` -- string - If not None, override the
       automatic choice of implementation. May be 'long' or 'vector',
```
Treat line 66?

kwankyu commented on src/sage/modules/vector_double_dense.pyx:
```diff
@@ -367,7 +367,7 @@ cdef class Vector_double_dense(Vector_numpy_dense):
 
-        - ``p`` - default: 2 - controls which norm is computed,
+        - ``p`` -- default: 2 - controls which norm is computed,
           allowable values are any real number and positive and
```
```
- ``p`` -- (default: 2) controls which norm is computed,
```

kwankyu commented on src/sage/rings/integer.pyx:
```diff
@@ -3860,28 +3860,28 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
 
         -  ``algorithm`` - string
 
-           - ``'pari'`` - (default) use the PARI library
```
line 3861?

kwankyu commented on src/sage/rings/polynomial/laurent_polynomial_mpair.pyx:
```diff
@@ -1734,7 +1734,7 @@ cdef class LaurentPolynomial_mpair(LaurentPolynomial):
 
-        - ``root`` - boolean (default ``False``) - if set to ``True``
+        - ``root`` -- boolean (default ``False``) - if set to ``True``
           then return a pair ``(True, sqrt)`` with ``sqrt`` a square
```
```
- ``root`` -- (default ``False``) if set to ``True``
```

kwankyu commented on src/sage/rings/polynomial/multi_polynomial_ideal.py:
```diff
@@ -1349,12 +1349,12 @@ def _groebner_basis_ginv(self, algorithm="TQ", criteria='CritPartially', divisio
                         - "C1", "CritPartially" (partial involutive criteria)
                         - "C1C2C3", "C1C2C3C4" (full involutive criteria)
 
-        - ``division_interface`` - either "Janet" or "JanetLike"
```
Together with line 1354,1455, manual editing is needed.

kwankyu commented on src/sage/rings/polynomial/pbori/pbori.pyx:
```diff
@@ -260,10 +260,10 @@ cdef class BooleanPolynomialRing(BooleanPolynomialRing_base):
 
     -  ``n`` - number of variables (an integer > 1)
 
-    - ``names`` - names of ring variables, may be a string or
```
line 261?

kwankyu commented on src/sage/rings/polynomial/pbori/pbori.pyx:
```diff
@@ -4817,7 +4817,7 @@ class BooleanPolynomialIdeal(MPolynomialIdeal):
 
         -  ``gens`` - a list of generators
 
-        - ``coerce`` - coerce all elements to the ring ``ring`` (default: ``True``)
```
line 4818?

kwankyu commented on src/sage/rings/polynomial/polynomial_zmod_flint.pyx:
```diff
@@ -176,7 +176,7 @@ cdef class Polynomial_zmod_flint(Polynomial_template):
 
-        - ``x`` - a list of coefficients - the coefficients are assumed to be
+        - ``x`` -- a list of coefficients - the coefficients are assumed to be
           reduced already and the list contains no trailing zeroes.
```
Maybe replace ` - ` with `;`. Not sure about this...

kwankyu commented on src/sage/schemes/cyclic_covers/cycliccover_generic.py:
```diff
@@ -51,9 +51,9 @@ def __init__(self, AA, r, f, names=None):
+        - ``A`` -- ambient affine space
 
-        - ``r`` - degree of the cover
+        - ``r`` -- degree of the cover
```
line 58?

kwankyu commented on src/sage/libs/singular/option.pyx:
```diff
@@ -476,26 +476,26 @@ cdef class LibSingularVerboseOptions(LibSingularOptions_abstract):
      - ``contentSB`` or ``content_sb`` - avoids to divide by the
        content of a polynomial in ``std`` and related algorithms.
        Should usually not be used.
-     - ``cancelunit`` - avoids to divide polynomials by non-constant
```
Untreated items above?

kwankyu commented on src/sage/categories/groups.py:
```diff
@@ -224,7 +224,7 @@ def cayley_table(self, names='letters', elements=None):
 
-            - ``elements`` - default = ``None``.  A list of
+            - ``elements`` -- default = ``None``.  A list of
               elements of the group, in forms that can be
```
```
- ``elements`` -- (default: ``None``)  a list of
```

kwankyu commented on src/sage/graphs/graph_generators.py:
```diff
@@ -1543,10 +1543,10 @@ def fullerenes(self, order, ipr=False):
 
-        - ``ipr`` - default: ``False`` - if ``True`` only fullerenes that
+        - ``ipr`` -- default: ``False`` - if ``True`` only fullerenes that
           satisfy the Isolated Pentagon Rule are generated. This means that
```
```
- ``ipr`` -- (default: ``False``) if ``True`` only fullerenes that
```

kwankyu replied:
Many similar changes needed in this file...

mkoeppe committed:
sage.categories: Update # needs, use block tags

mkoeppe committed:
sage.combinat: Modularization fixes (imports), update # needs

mkoeppe committed:
src/sage/algebras/lie_algebras/bch.py: Fix import of IntegerListsLex

gmou3 committed:
Double dashes in more peculiar patterns

fchapoton committed:
remove inheritance from Algebra in FreeAlgebras

gmou3 committed:
Suggestion for manual change by kwankyu

gmou3 committed:
Add dashes and simplify (optional, default:*)

gmou3 committed:
Merge tag '10.4.beta7' into setsystem_input

SageMath version 10.4.beta7, Release Date: 2024-05-25

fchapoton committed:
fix pep8 E302 in algebras and categories (scripted)

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: fix typo and small doc update

DavidAyotte committed:
src/sage/modular/drinfeld_modform/ring.py: remove redundant begining index in range function

DavidAyotte committed:
src/sage/modular/drinfeld_modform/element.py: return NotImplemented for invalid operator

fchapoton committed:
fixing superfluous empty lines in libs folder

DavidAyotte committed:
Merge branch 'develop' into drinfeld_modform

DavidAyotte committed:
src/sage/modular/drinfeld_modform/all.py: replace relative import for an absolute import

mkoeppe committed:
for a in "CLASS='ModularAbelianVariety';SUFFIX=_abstract" "CLASS='ArithmeticSubgroup|ModularFormsSpace|ModularSymbolsSpace';SUFFIX=" "CLASS='Gamma|Gamma0|Gamma1|GammaH|SL2Z|DirichletGroup';SUFFIX='_class'" "CLASS='CongruenceSubgroup';SUFFIX='Base'"; do eval "$a"; git --no-pager grep -l -E 'is_('$CLASS')' | xargs sed -E -i.bak 's/import is_('$CLASS')/import \1'$SUFFIX'/;/^ +/s/is_('$CLASS')[(]([^()]*([(][^()]*[)])?[^()]*)[)]/isinstance(\2, \1'$SUFFIX')/g'; done

mkoeppe committed:
fixups

mkoeppe committed:
LPDictionary: Make it safe to copy dictionaries

antonio-rojas committed:
Fix sage.symbolic feature after libs/pynac removal

fchapoton committed:
Merge branch 'develop' into simplify_free_algebra

kwankyu committed:
Run script in /sage

vbraun committed:
Still add rpath to our own libstdc++, in case we built our own toolchain

mkoeppe committed:
src/sage/modular/arithgroup/all.py: Fix rebase

mkoeppe committed:
src/sage/misc/dev_tools.py: Update doctest output

kwankyu commented on src/sage/algebras/letterplace/free_algebra_letterplace.pyx:
```diff
@@ -644,8 +644,8 @@ cdef class FreeAlgebra_letterplace(Algebra):
-        ``d`` - an integer.
+        ``g`` -- a list of elements of this free algebra.
+        ``d`` -- an integer.
 
```
leading `-`?

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra.py:
```diff
@@ -520,12 +520,12 @@ def __init__(self, p=2, basis='milnor', **kwds):
-        - ``generic`` -- (optional, default 'auto')
+        - ``p`` -- positive prime integer (default: 2)
+        - ``basis`` -- string (default: = 'milnor')
+        - ``profile`` -- profile function (default: ``None``)
```
```
- ``basis`` -- string (default: ``'milnor'``)
```
Similarly below.

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra.py:
```diff
@@ -2314,7 +2314,7 @@ def P(self, *nums):
 
-        -  ``a, b, c, ...`` - non-negative integers
+        -  ``a, b, c, ...`` -- non-negative integers
 
```
```
- ``a``, ``b``, ``c``, ... -- non-negative integers
```

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra.py:
```diff
@@ -3807,12 +3807,12 @@ def SteenrodAlgebra(p=2, basis='milnor', generic='auto', **kwds):
-    - ``generic`` -- (optional, default 'auto')
+    - ``p`` -- positive prime integer (default: = 2)
+    - ``basis`` -- string (default: = 'milnor')
+    - ``profile`` -- a profile function in form specified below (default: ``None``)
```
The same comment...

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra.py:
```diff
@@ -4226,7 +4226,7 @@ def Sq(*nums):
 
-    -  ``a, b, c, ...`` - non-negative integers
+    -  ``a, b, c, ...`` -- non-negative integers
 
```
The same comment

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra_misc.py:
```diff
@@ -653,12 +653,12 @@ def serre_cartan_mono_to_string(mono, latex=False, generic=False):
 
-    - ``latex`` -- boolean (optional, default False), if true, output
+    - ``latex`` -- boolean (default: False), if true, output
       LaTeX string
```
```
``False``
```
But perhaps this change is out of scope of your PR. I am okay if you do not do that here. If you want to do it in the lines you touched, there are many more.

gmou3 replied:
Done.

kwankyu commented on src/sage/algebras/steenrod/steenrod_algebra_mult.py:
```diff
@@ -697,9 +697,9 @@ def adem(a, b, c=0, p=2, generic=None):
 
-    - `a`, `b`, `c` (optional) - nonnegative integers, corresponding
+    - `a`, `b`, `c` (optional) -- nonnegative integers, corresponding
       to either `P^a P^b` or (if `c` present) to `P^a \beta^b P^c`
```
```
``a``
```
But out of the scope of your PR.

gmou3 replied:
I'll leave these for another PR.

kwankyu commented on src/sage/calculus/all.py:
```diff
@@ -31,204 +31,4 @@
-    return SR(x)
-
-
 from . import desolvers
```
???

gmou3 replied:
This is incoming from develop.

kwankyu commented on src/sage/coding/databases.py:
```diff
@@ -197,7 +197,7 @@ def self_orthogonal_binary_codes(n, k, b=2, parent=None, BC=None, equal=False,
 
-    -  ``parent``- - Used in recursion (default: ``None``)
+    -  ``parent``- -- Used in recursion (default: ``None``)
 
```
needs manual touch.

kwankyu commented on src/sage/combinat/cluster_algebra_quiver/mutation_class.py:
```diff
@@ -197,7 +197,7 @@ def _dg_canonical_form(dg, frozen=None):
 
-    - ``frozen`` -- list (optional, default []) of frozen vertices
+    - ``frozen`` -- list (default: []) of frozen vertices
 
```
```
``[]``
```

kwankyu commented on src/sage/combinat/designs/block_design.py:
```diff
@@ -210,7 +210,7 @@ def ProjectiveGeometryDesign(n, d, F, algorithm=None, point_coordinates=True, ch
 
-    - ``check`` -- (optional, default to ``True``) whether to check the output.
+    - ``check`` -- (default: to ``True``) whether to check the output.
 
```
remove `to`

kwankyu commented on src/sage/combinat/designs/twographs.py:
```diff
@@ -103,7 +103,7 @@ def is_regular_twograph(self, alpha=False):
 
-        - ``alpha`` -- (optional, default is ``False``) return the value of
+        - ``alpha`` -- (default: is ``False``) return the value of
           ``alpha``, if possible.
```
remove `is`.

kwankyu commented on src/sage/combinat/subword_complex.py:
```diff
@@ -1964,9 +1964,9 @@ def _greedy_facet(Q, w, side="negative", n=None, pos=0, l=None, elems=[]):
+    - ``n`` -- an integer (default:s to the length of `Q`)
+    - ``pos`` -- an integer (default: 0)
+    - ``l`` -- an integer (default:s to the length of `w`)
     - ``elems`` -- a list (optional)
```
neeeds manual touch...

kwankyu commented on src/sage/graphs/digraph_generators.py:
```diff
@@ -666,7 +666,7 @@ def nauty_directg(self, graphs, options="", debug=False):
 
-        - ``debug`` (boolean) -- default: ``False`` - if ``True``
+        - ``debug`` (boolean) -- (default: ``False``); if ``True``
           directg standard error and standard output are displayed.
```
There are many cases like this. It seems to me that the semicolon is redundant.

kwankyu commented on src/sage/interfaces/fricas.py:
```diff
@@ -1705,12 +1705,12 @@ def _sage_expression(fricas_InputForm):
                 |       2                    2
-             - \|- 3 %y1  - 2 %y0 %y1 - 3 %y0   - %y1 - %y0
+             - \|- 3 %y1  - 2 %y0 %y1 - 3 %y0   - %y1 -- %y0
              ----------------------------------------------]
```
Revert!

gmou3 replied:
:O

kwankyu commented on src/sage/interfaces/gnuplot.py:
```diff
@@ -64,15 +64,15 @@ def plot(self, cmd, file=None, verbose=True, reset=True):
         INPUT:
 
 
-        -  ``cmd`` - string
```
redundant blank line above.

kwankyu commented on src/sage/interfaces/kash.py:
```diff
@@ -542,13 +542,13 @@ def eval(self, x, newlines=False, strip=True, **kwds):
         INPUT:
 
 
-        -  ``s`` - string containing Kash code.
```
redundant blank line above.

kwankyu commented on src/sage/interfaces/magma.py:
```diff
@@ -420,13 +420,13 @@ def _read_in_file_command(self, filename):
 
-        -  ``filename`` - string
+        -  ``filename`` -- string
 
```
redundant blank lines above and below.

kwankyu commented on src/sage/interfaces/maple.py:
```diff
@@ -751,7 +751,7 @@ def source(self, s):
 
 
-        -  ``s`` - a string representing the function whose
+        -  ``s`` -- a string representing the function whose
```
redundant blank lines.

kwankyu commented on src/sage/lfunctions/lcalc.py:
```diff
@@ -211,18 +211,18 @@ def values_along_line(self, s0, s1, number_samples, L=''):
         OUTPUT:
 
 
-        -  ``list`` - list of pairs (s, zeta(s)), where the s
```
redundant blank line above and below

kwankyu commented on src/sage/lfunctions/sympow.py:
```diff
@@ -247,15 +247,15 @@ def analytic_rank(self, E):
+        -  ``integer`` -- analytic rank
 
-        -  ``string`` - leading coefficient (as string)
+        -  ``string`` -- leading coefficient (as string)
```
Wonder why there are many blank lines...

kwankyu commented on src/sage/matrix/matrix2.pyx:
```diff
@@ -5981,15 +5981,15 @@ cdef class Matrix(Matrix1):
         OUTPUT:
 
 
-        -  ``list`` - list of Vectors
```
blank lines again.

kwankyu commented on src/sage/matrix/matrix_gfpn_dense.pyx:
```diff
@@ -873,7 +873,7 @@ cdef class Matrix_gfpn_dense(Matrix_dense):
         - `i`: Index of the first row to be extracted
-        - `j` (optional, default -1): -1, or index of the last
+        - `j` (default: -1): -1, or index of the last
           row to be extracted.
```
`--`?

kwankyu commented on src/sage/matrix/matrix_integer_dense.pyx:
```diff
@@ -2131,25 +2131,25 @@ cdef class Matrix_integer_dense(Matrix_dense):
         OUTPUT:
 
 
-        -  ``matrix`` - a matrix over ZZ
```
Perhaps removing these redundant blank lines should be a separate PR.

kwankyu commented on src/sage/misc/sphinxify.py:
```diff
@@ -39,7 +39,7 @@ def sphinxify(docstring, format='html'):
 
-    - ``format`` -- string (optional, default 'html') -- either 'html' or
+    - ``format`` -- string (default: 'html') -- either 'html' or
       'text'
```
`;`?

kwankyu commented on src/sage/rings/polynomial/multi_polynomial_ideal.py:
```diff
@@ -3037,7 +3037,7 @@ def hilbert_series(self, grading=None, algorithm='sage'):
              - t^10 - t^9 - t^8 - t^7 - t^6 - t^5 - t^4 - t^3 - t^2
-             - t - 1)/(t^12 + t^11 + t^10 - t^2 - t - 1)
+             - t -- 1)/(t^12 + t^11 + t^10 - t^2 - t - 1)
 
```
revert.

kwankyu commented on src/sage/schemes/elliptic_curves/gal_reps.py:
```diff
@@ -1108,7 +1108,7 @@ def image_classes(self,p,bound=10000):
 
-        - a natural number ``bound`` (optional, default=10000)
+        - a natural number ``bound`` (default:=10000)
 
```
`: `

kwankyu commented on src/sage/schemes/generic/algebraic_scheme.py:
```diff
@@ -839,10 +839,10 @@ def rational_points(self, **kwds):
 
-        - ``bound`` - integer (optional, default=0). The bound for the coordinates for
+        - ``bound`` -- integer (default:=0). The bound for the coordinates for
           subschemes with dimension at least 1.
```
`: ` here and below

kwankyu commented on src/sage/schemes/projective/projective_homset.py:
```diff
@@ -486,7 +486,7 @@ def points(self, B=0):
 
-        - ``B`` -- integer (optional, default=0). The bound for the
+        - ``B`` -- integer (default:=0). The bound for the
           coordinates.
```
`: `?

kwankyu commented on src/sage/schemes/projective/projective_point.py:
```diff
@@ -1063,7 +1063,7 @@ class SchemeMorphism_point_projective_field(SchemeMorphism_point_projective_ring
 
-    - ``check`` -- boolean (optional, default:``True``). Whether to
+    - ``check`` -- boolean (default:``True``). Whether to
       check the input for consistency.
```
`: ` (blank)

kwankyu commented on src/sage/schemes/toric/morphism.py:
```diff
@@ -790,7 +790,7 @@ class SchemeMorphism_fan_toric_variety(SchemeMorphism, Morphism):
 
-    - ``check`` -- boolean (optional, default:``True``). Whether to
+    - ``check`` -- boolean (default:``True``). Whether to
       check the input for consistency.
```
`: ` (blank)

fchapoton committed:
fix error made in src/sage/libs/giac/giac.pyx

Co-authored-by: Matthias Köppe <mkoeppe@math.ucdavis.edu>

vbraun committed:
Fix doctest when the optional benzene is installed

gmou3 committed:
Format around "(default: )" and add preceding dash

kwankyu commented on src/sage/rings/polynomial/infinite_polynomial_ring.py:
```diff
@@ -522,8 +522,8 @@ def __init__(self, parent, start):
-        ``parent`` -- a ring.
-        ``start`` -- some dictionary, usually the dictionary of variables of ``parent``.
+        - ``parent`` -- a rin
+        - ``start`` -- some dictionary, usually the dictionary of variables of ``parent``.
```
`a ring.`

kwankyu commented on src/sage/rings/polynomial/infinite_polynomial_ring.py:
```diff
@@ -669,9 +669,9 @@ def __init__(self, R, names, order):
-        ``names`` -- list of generator names.
-        ``order`` -- string determining the monomial order of the infinite polynomial ring.
+        - ``R`` -- base rin
+        - ``names`` -- list of generator name
```
`base ring`

`generator names`

kwankyu commented on src/sage/rings/polynomial/symmetric_ideal.py:
```diff
@@ -184,9 +184,9 @@ def __init__(self, ring, gens, coerce=True):
-        ``coerce`` -- (bool, default ``True``) coerce the given generators into ``ring``
+        - ``ring`` -- an infinite polynomial ri
+        - ``gens`` -- generators of this ide
+        - ``coerce`` -- (bool, default ``True``) coerce the given generators into ``ring``
```
`ring`

`ideal`

gmou3 committed:
Correcting a few chop-offs

gmou3 committed:
`doc`: Update `help()` outputs

kwankyu reviewed (approved):
Thanks.

Commit:
gh-36538: Implement Drinfeld modular forms
    
<!-- ^^^^^
Please provide a concise, informative and self-explanatory title.
Don't put issue numbers in there, do this in the PR body below.
For example, instead of "Fixes #1234" use "Introduce new method to
calculate 1+1"
-->
<!-- Describe your changes here in detail -->

<!-- Why is this change required? What problem does it solve? -->
<!-- If this PR resolves an open issue, please link to it here. For
example "Fixes #12345". -->
<!-- If your change requires a documentation PR, please link it
appropriately. -->

This PR aims to implement Drinfeld modular forms into SageMath. These
special types of modular forms live in the function field theory and are
linked with Drinfeld modules.

#### Definitions

Without going too much into the details, we give some definitions and
insight of what is going on here. Let $K = \mathbb{F}_q(T)$ together
with the $1/T$-adic norm and $A = \mathbb{F}_q[T]$. We define
$K\_{\infty} = \mathbb{F}_q((1/T))$ be the completion of $K$ and let
$\mathbb{C}\_{\infty}$ be the completion of an algebraic closure of
$K\_{\infty}$. In this specific setting, the *Drinfeld period domain* of
rank $r$ is

$$
\Omega^r(\mathbb{C}\_{\infty}) := \lbrace (w_1, \ldots, w_{r-1}, 1)\in
\mathbb{C}\_{\infty}^r : \text{the }w_i \text{'s are
}K\_{\infty}\text{-linearly independant} \rbrace.
$$

(this period domain is to be seen as the analogue of the complex upper
half plane)

So, for each $w = (w_1, \ldots, w_{r-1}, 1)\in
\Omega^r(\mathbb{C}\_{\infty})$, we have an associated $A$-lattice (i.e.
a discrete projective $A$-module of rank r): $\Lambda_w = A w_1 + \ldots
A w_{r-1} + A$.

The analytic theory of Drinfeld modules tells us that, for each such
$w$, there exist a Drinfeld module

$$
\phi^w_T : T \mapsto T + g\_1(w) \tau + \ldots + g\_{r-1}(w) \tau^{r-1}
+ g\_r(w) \tau^r
$$

associated with this lattice. This association is what we call the
universal Drinfeld module over $\Omega^r(\mathbb{C}\_{\infty})$. We see
the coefficients as functions $g_i : \Omega^r(\mathbb{C}\_{\infty})
\rightarrow \mathbb{C}\_{\infty}$. These functions satisfy modular
transformation property under the action of the group
$\mathrm{GL}_r(A)$:

$$
g_i(\gamma(w)) = j(\gamma, w)^i g_i(w)
$$

where $j$ is some automorphic factor.

The ring of Drinfeld modular forms for $\mathrm{GL}_r(A)$ is

$$
\mathcal{M}\_{\bullet} = \mathbb{C}\_{\infty}[g_1, \ldots g_{r-1}, g_r].
$$

#### Implementations details

In this implementation, we create a new parent `DrinfeldModularForms`
which represents the polynomial ring $K[g_1,\ldots, g_{r-1}, g_r]$ and a
new element class `DrinfeldModularFormsElement`. Then, for arbitrary
rank, we only see a Drinfeld modular form symbolically as a multivariate
polynomial in $g_1, \ldots, g_r$, the coefficients forms. In the special
case when the rank is two, we aim to implement algorithms for computing
the expansion at infinity of any Drinfeld modular forms. These
expansions are analogues of the usual $q$-expansion of classcial modular
forms. This will be achieved via the Lopez-Petrov nonstandard expansion
theory.

We note that this implementation was initially done as a pip-installable
package:

https://davidayotte.github.io/drinfeld_modular_forms

This work has been an ongoing project for almost two years now and since
the addition of Drinfeld modules in SageMath, the author has seen an
interest in the implementation of Drinfeld modular forms to SageMath. In
the rank two case, the ability to compute their expansion at infinity
shows some interesting computational properties of these objects.

We are aware that this is a quite specialized topic, so we aim to make
this implementation as convivial as possible for the user and well
documented. Any comments and suggestions are welcome!

CC: @xcaruso @kryzar @ymusleh

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it
appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're
here to help! -->
<!-- Feel free to remove irrelevant items. -->

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on
- #12345: short description why this is a dependency
- #34567: ...
-->

<!-- If you're unsure about any of these, don't hesitate to ask. We're
here to help! -->
    
URL: https://github.com/sagemath/sage/pull/36538
Reported by: David Ayotte
Reviewer(s): Antoine Leudière, David Ayotte, Xavier Caruso

Commit:
gh-37848: Implement all G(r,p,m) complex reflection groups
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

We provide an implementation of the whole family of `G(r, p, n)` complex
reflection groups by realizing them as a subgroup of
`ColoredPermutations`.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/37848
Reported by: Travis Scrimshaw
Reviewer(s): Frédéric Chapoton

Commit:
gh-37866: LPDictionary: Make it safe to copy dictionaries
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

Dictionaries are mutable, and copying them using `copy` results in
dictionaries that share internal objects, which causes unwelcome
surprises.

- rebased from #31308

Author: @mkoeppe, @ComboProblem

Fixes #31308.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/37866
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee, Matthias Köppe

Commit:
gh-37919: Upgrades and configure fixes for macOS arm64
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

Also
- update for updated GitHub macOS runner labels

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

- Depends on #37726 (merged here to resolve merge conflict)
    
URL: https://github.com/sagemath/sage/pull/37919
Reported by: Matthias Köppe
Reviewer(s):

Commit:
gh-37999: Fix broken pytest integration (including CI failure in `src/conftest_test.py`)
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

Frequently seen in the CI, for example https://github.com/sagemath/sage/
actions/runs/9055116851/job/24875749181?pr=37862#step:12:27
```
Doctesting files changed since last git commit
No files to doctest
Features detected for doctesting:
Running pytest on ['/sage/src'] with options []
============================= test session starts
==============================
platform linux -- Python 3.10.12, pytest-8.2.0, pluggy-1.5.0
rootdir: /sage/src
configfile: tox.ini
plugins: xdist-3.6.1, mock-3.14.0, anyio-4.0.0, hypothesis-6.100.7
collected 63 items / 2 skipped

src/conftest_test.py .F
[  3%]
src/sage/doctest/parsing_test.py ...........
[ 20%]
src/sage/manifolds/differentiable/examples/symplectic_space_test.py ....
[ [26](https://github.com/sagemath/sage/actions/runs/9055116851/job/2487
5749181?pr=37862#step:12:27)%]
 [ 26%]
src/sage/manifolds/differentiable/symplectic_form_test.py ..............
[ 49%]
.....................
[ 82%]
src/sage/manifolds/differentiable/tensorfield_paral_test.py .
[ 84%]
src/sage/numerical/backends/cvxopt_backend_test.py ..
[ 87%]
src/sage/numerical/backends/glpk_backend_test.py ..
[ 90%]
src/sage/numerical/backends/glpk_exact_backend_test.py ..
[ 93%]
src/sage/numerical/backends/interactivelp_backend_test.py ..
[ 96%]
src/sage/numerical/backends/ppl_backend_test.py ..
[100%]

=================================== FAILURES
===================================
______________ TestPytestSageScript.test_invoke_on_inputtest_file
______________

self = <conftest_test.TestPytestSageScript object at 0x7f0d91ecdae0>

    def test_invoke_on_inputtest_file(self):
        result = subprocess.run(
            ["sage", "--pytest", input_file],
            capture_output=True,
            text=True,
        )
>       assert result.returncode == 1  # There are failures in the input
test
E       assert 2 == 1
E        +  where 2 = CompletedProcess(args=['sage', '--pytest',
PosixPath('/sage/src/conftest_inputtest.py')], returncode=2,
stdout="======...n !!!!!!!!!!!!!!!!!!!!\n===============================
1 error in 0.18s ===============================\n",
stderr='').returncode

src/conftest_test.py:40: AssertionError
=========================== short test summary info
============================
FAILED src/conftest_test.py::TestPytestSageScript::test_invoke_on_inputt
est_file
============== 1 failed, 62 passed, 2 skipped in 97.56s (0:01:[37](https
://github.com/sagemath/sage/actions/runs/9055116851/job/24875749181?pr=3
7862#step:12:38)) ==============Running pytest on ['/sage/src'] with
options []
============================= test session starts
==============================
platform linux -- Python 3.10.12, pytest-8.2.0, pluggy-1.5.0
rootdir: /sage/src
configfile: tox.ini
plugins: xdist-3.6.1, mock-3.14.0, anyio-4.0.0, hypothesis-6.100.7
collected 63 items / 2 skipped

src/conftest_test.py .F
[  3%]
src/sage/doctest/parsing_test.py ...........
[ 20%]
src/sage/manifolds/differentiable/examples/symplectic_space_test.py ....
[ 26%]
 [ 26%]
src/sage/manifolds/differentiable/symplectic_form_test.py ..............
[ 49%]
.....................
[ 82%]
src/sage/manifolds/differentiable/tensorfield_paral_test.py .
[ 84%]
src/sage/numerical/backends/cvxopt_backend_test.py ..
[ 87%]
src/sage/numerical/backends/glpk_backend_test.py ..
[ 90%]
src/sage/numerical/backends/glpk_exact_backend_test.py ..
[ 93%]
src/sage/numerical/backends/interactivelp_backend_test.py ..
[ 96%]
src/sage/numerical/backends/ppl_backend_test.py ..
[100%]

=================================== FAILURES
===================================
______________ TestPytestSageScript.test_invoke_on_inputtest_file
______________

self = <conftest_test.TestPytestSageScript object at 0x7f0d91ecdae0>

    def test_invoke_on_inputtest_file(self):
        result = subprocess.run(
            ["sage", "--pytest", input_file],
            capture_output=True,
            text=True,
        )
>       assert result.returncode == 1  # There are failures in the input
test
E       assert 2 == 1
E        +  where 2 = CompletedProcess(args=['sage', '--pytest',
PosixPath('/sage/src/conftest_inputtest.py')], returncode=2,
stdout="======...n !!!!!!!!!!!!!!!!!!!!\n===============================
1 error in 0.18s ===============================\n",
stderr='').returncode

src/conftest_test.py:40: AssertionError
=========================== short test summary info
============================
FAILED src/conftest_test.py::TestPytestSageScript::test_invoke_on_inputt
est_file
============== 1 failed, 62 passed, 2 skipped in 97.56s (0:01:[37](https
://github.com/sagemath/sage/actions/runs/9055116851/job/24875749181?pr=3
7862#step:12:38)) ==============
```

- Fixes #37536.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/37999
Reported by: Matthias Köppe
Reviewer(s): Nathan Dunfield

Commit:
gh-38009: `build/pkgs/ecl`: Update to 24.5.10
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Fixes #38004
- Fixes #37511 (?)

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38009
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee

Commit:
gh-38032: `sage.combinat.finite_state_machine`: Deprecate `is_...` functions
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->



### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38032
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee, Matthias Köppe

Commit:
gh-38035: `sage.modular`: Deprecate `is_...` functions
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->



### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38035
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee, Matthias Köppe

Commit:
gh-38056: `SetSystem`: Minor change to accommodate set input
    
This change makes the code cleaner in multiple places.
The PR also includes some docstring edits in `set_system.pyx`.

### :hourglass: Dependencies

- Depends on #37930 and #38027
    
URL: https://github.com/sagemath/sage/pull/38056
Reported by: gmou3
Reviewer(s): Matthias Köppe

Commit:
gh-38059: `sage.ext.fast_callable`: Docstring cosmetics
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Cherry-picked from #35095

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38059
Reported by: Matthias Köppe
Reviewer(s): gmou3, Kwankyu Lee, Matthias Köppe

Commit:
gh-38060: `sage.rings`: Modularization fixes for singular
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

Removing the use of a Singular interpreter instance as a default value
of parameters

- Cherry-picked from #35095.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38060
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee, Matthias Köppe

Commit:
gh-38061: `sage.rings`: Modularization fixes (imports)
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

Replacing unconditional imports from `sage.libs.pari`,
`sage.misc.sage_eval`, `sage.rings.number_field`,
`sage.rings.polynomial.pbori`, `sage.symbolic` using standard
modularization techniques.

- Cherry-picked from #35095

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38061
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee

Commit:
gh-38065: `sage.graphs`: Doctest cosmetics
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Cherry-picked from #35095.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38065
Reported by: Matthias Köppe
Reviewer(s): gmou3

Commit:
gh-38066: `sage.geometry`: Modularization fixes (imports), `# needs`
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Cherry-picked from #35095

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38066
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee

Commit:
gh-38067: `.ci/write-dockerfile.sh`: Quoting fix for `unminimize`
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

On docker-ubuntu, we attempted to remove the `unminimize` program after
running it once, but this has failed because of  quoting mistake. https:
//github.com/mkoeppe/sage/actions/runs/9219405769/job/25364502464#step:9
:311

Fixed here.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38067
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee, Matthias Köppe

Commit:
gh-38071: `sage.categories`: Update `# needs`, use block tags
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Cherry-picked from #35095.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38071
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee

Commit:
gh-38074: `sage.combinat`: Modularization fixes (imports), update `# needs`
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

- Cherry-picked from #35095.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38074
Reported by: Matthias Köppe
Reviewer(s): Kwankyu Lee

Commit:
gh-38078: remove inheritance from Algebra in FreeAlgebras
    
as another move towards removal of the "Algebra" class, using "Parent"
framework instead.

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.
    
URL: https://github.com/sagemath/sage/pull/38078
Reported by: Frédéric Chapoton
Reviewer(s): Matthias Köppe

Commit:
gh-38079: fix pep8 E302 in algebras and categories (scripted)
    
This fixes all warnings for pep8 E302 in `algebras` and `categories`
folders :
```
E302 expected 2 blank lines, found 0 or 1.
```

This was purely scripted using `autopep8`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
    
URL: https://github.com/sagemath/sage/pull/38079
Reported by: Frédéric Chapoton
Reviewer(s): Matthias Köppe

Commit:
gh-38080: fixing superfluous empty lines in libs folder
    
this is removing empty lines in the `libs` folder in `.pyx` files, for
compliance with pep8 E302 and E305

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
    
URL: https://github.com/sagemath/sage/pull/38080
Reported by: Frédéric Chapoton
Reviewer(s): Matthias Köppe

Commit:
gh-38081: Fix sage.symbolic feature after libs/pynac removal
    
Fixes
https://github.com/sagemath/sage/pull/37873#issuecomment-2131371458
    
URL: https://github.com/sagemath/sage/pull/38081
Reported by: Antonio Rojas
Reviewer(s):

Commit:
gh-38082: CI fix: broken livedoc
    
<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR
description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method
to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For
example, "Fixes #12345". -->

as seen in

https://github.com/sagemath/sage/actions/runs/9234699215/job/25408704331
#:~:text=bash%3A%20line%201%3A%20cd%3A%20./worktree%2Dimage%3A%20No%20su
ch%20file%20or%20directory

### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation
preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example,
-->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->
    
URL: https://github.com/sagemath/sage/pull/38082
Reported by: Kwankyu Lee
Reviewer(s):

Commit:
gh-38087: Still add rpath to our own libstdc++, in case we built our own toolchain
    
The GMP configure test should link against our own libstdc++ instead of
the (too old) system libstdc++


Fixes #38085
    
URL: https://github.com/sagemath/sage/pull/38087
Reported by: Volker Braun
Reviewer(s):

Commit:
gh-38091: Fix doctest when the optional benzene is installed
    
```
**********************************************************************
File "src/sage/features/__init__.py", line 384, in
sage.features.Feature.hide
Failed example:
    len(list(graphs.fusenes(2)))  # optional - benzene, needs
sage.graphs
Expected nothing
Got:
    1
**********************************************************************
1 item had failures:
   1 of   6 in sage.features.Feature.hide
    [146 tests, 1 failure, 3.74 s]
**********************************************************************
```

Comes from 1ade282122269d5369d53c07cdf069984f89bd8a
    
URL: https://github.com/sagemath/sage/pull/38091
Reported by: Volker Braun
Reviewer(s):

Commit:
gh-38092: `doc`: Update `help()` outputs

### :memo: Checklist

- [x] The title is concise and informative.
- [x] I have updated the documentation and checked the documentation
preview.

URL: https://github.com/sagemath/sage/pull/38092
Reported by: gmou3
Reviewer(s): Matthias Köppe
